src/main/java/ch/uzh/ifi/hase/soprafs22/game/Game.java

```diff
@@ -76,4 +78,86 @@ public boolean hasEnded(){
     public boolean isPlayersTurn(String token){
         return turnOrder[turnNumber % turnOrder.length].equals(token);
     }
+
+    public void attack(String token, Position attacker, Position defender) throws NotPlayersTurnException,
+            TileOutOfRangeException,
+            AttackOutOfRangeException,
+            NotAMemberOfGameException,
+            GameOverException,
+            UnitNotFoundException,
+            WrongUnitOwnerException,
+            WrongTargetTeamException {
+        ensureMember(token);
+        ensureNotEnded();
+        ensureTurn(token);
+        Unit attackingUnit = gameMap.getTile(attacker).getUnit();
```

Comment:
i changed that, but i'm open to let me convince me otherwise.

Reply:
It is still usable, but finding units gets more complicated. For example, when checking if unit at position (x,y) belongs to an enemy or teammate we used to:
1. Get tile at (x, y).
2. Check the team of the unit.

Now we have to:
1. Find all players in the game.
2. Find all units of each player.
3. Check if one is at position (x,y).
4. Check the team of that unit.

I've changed it to the current implementation, please fix this if you decide to keep units in their tiles.

Reply:
Good thought. I'll think about this.